search-impl/src/test/java/com/example/auction/search/impl/SearchServiceImplTest.java

```diff
@@ -0,0 +1,273 @@
+        itemStub.get().tell(itemCreated1, ActorRef.noSender());
+    }
+
+    private void startAuction(UUID itemId1, UUID creatorId, int reservePrice, int increment) {
+        ItemEvent.AuctionStarted auctionStarted1 = new ItemEvent.AuctionStarted(itemId1, creatorId, reservePrice, increment, Instant.now(), Instant.now().plusSeconds(50));
+        itemStub.get().tell(auctionStarted1, ActorRef.noSender());
+    }
+
+    private void assertCountOfResults(int expectedCount, UUID itemId1) {
+        assertCountOfResults(expectedCount, itemId1, new SearchRequest(Optional.empty(), Optional.empty(), Optional.empty()));
+    }
+
+    private void assertCountOfResults(int expectedCount, UUID itemId1, SearchRequest request) {
+        eventually(new FiniteDuration(10, TimeUnit.SECONDS), new FiniteDuration(1, TimeUnit.SECONDS), () -> {
+            SearchResult items = searchService
+                    .search(0, 15)
+                    .invoke(request)
+                    .toCompletableFuture()
+                    .get(10, SECONDS);
+            assertEquals(expectedCount, items.getItems().size());
+            if (expectedCount > 0) {
+                assertEquals(itemId1, items.getItems().get(0).getId());
+            }
+        });
+    }
+
+    // ------------------------------------------------------------------
+
+    public static class ItemStub implements ItemService {
+
+        private final TopicStub<ItemEvent> topicStub;
+
+        @Inject
+        public ItemStub(Materializer materializer) {
+            topicStub = new TopicStub<>(materializer);
+            itemStub = topicStub.actorSupplier();
+        }
+
+        @Override
+        public Topic<ItemEvent> itemEvents() {
+            return topicStub;
+        }
+
+        @Override
+        public ServiceCall<ItemData, Item> createItem() {
+            return null;
+        }
+
+        @Override
+        public ServiceCall<ItemData, Item> updateItem(UUID id) {
+            return null;
+        }
+
+        @Override
+        public ServiceCall<NotUsed, Done> startAuction(UUID id) {
+            return null;
+        }
+
+        @Override
+        public ServiceCall<NotUsed, Item> getItem(UUID id) {
+            return null;
+        }
+
+        @Override
+        public ServiceCall<NotUsed, PaginatedSequence<ItemSummary>> getItemsForUser(UUID id, ItemStatus status, Optional<Integer> pageNo, Optional<Integer> pageSize) {
+            return null;
+        }
+    }
+
+    // -----------------------------------------------------------------
+
+    public static class BiddingStub implements BiddingService {
+        private final TopicStub<BidEvent> topicStub;
+
+        @Inject
+        public BiddingStub(Materializer materializer) {
+            topicStub = new TopicStub<>(materializer);
+            bidStub = topicStub.actorSupplier();
+        }
```

Comment:
A Stubbed service can locally create as many stubs as Topic it will use. After creating the `TopicStub` the user may request the `actorSupplier` and assign it to a test field (see code above).